### ISSUE TITLE
Value 'sm_86' is not defined for option 'gpu-name'

### ISSUE THREAD

**Issue by liuxufenfeiya** [tensorflow/tensorflow#45590]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):docker
- TensorFlow installed from (source or binary):pip
- TensorFlow version (use command below):tf-nightly-gpu         2.5.0.dev20201210
- Python version:3.6.9
- CUDA/cuDNN version:11.0/8.0.4
- GPU model and memory:3090/24GB

when i inference my tf model,it appears hundreds of warnings like the following image 
about 1 minute later ,the normal inference begin
i watch the issus and try to used tf2.4re02 but it was useless
![image](https://user-images.githubusercontent.com/33931180/101855756-ef43c680-3b9e-11eb-807e-7756853c75ab.png)

**Comment by ravikyram** [comment]
@liuxufenfeiya 

You can ignore those warning.If you want to disable the warning messages add the below code.

```
import os
import tensorflow as tf
os.environ['TF_CPP_MIN_LOG_LEVEL'] = "2"

```
If you feel the issue still persists request you to share colab link or simple standalone code to reproduce the issue in our environment. It helps us in localizing the issue faster.
Thanks!

**Comment by liuxufenfeiya** [comment]
> @liuxufenfeiya
> 
> You can ignore those warning.If you want to disable the warning messages add the below code.
> 
> ```
> import os
> import tensorflow as tf
> os.environ['TF_CPP_MIN_LOG_LEVEL'] = "2"
> ```
> 
> If you feel the issue still persists request you to share colab link or simple standalone code to reproduce the issue in our environment. It helps us in localizing the issue faster.
> Thanks!

i had already try the way,but it was no use

**Comment by king398** [comment]
Try upgrading your Duda version to 11.1

**Comment by crutcher** [comment]
cuda 11.1 does not solve the problem.

```
2020-12-14 01:30:45.379556: W tensorflow/stream_executor/gpu/asm_compiler.cc:235] Your CUDA software stack is old. We fallback to the NVIDIA driver for some compilation. Update your CUDA version to get the best performance. The ptxas error was: ptxas fatal   : Value 'sm_86' is not defined for option 'gpu-name'
```

```
(venv)crutcher@heatlamp:~/git/smot$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.1 LTS
Release:	20.04
Codename:	focal
```

```
(venv)crutcher@heatlamp:~/git/smot$ nvidia-smi
Mon Dec 14 01:34:58 2020       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 455.38       Driver Version: 455.38       CUDA Version: 11.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  GeForce RTX 3090    Off  | 00000000:21:00.0  On |                  N/A |
| 30%   38C    P8    34W / 350W |  23683MiB / 24265MiB |     14%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   1  GeForce RTX 3090    Off  | 00000000:4A:00.0 Off |                  N/A |
| 30%   33C    P8    31W / 350W |    413MiB / 24268MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A      2618      G   /usr/lib/xorg/Xorg                555MiB |
|    0   N/A  N/A      2895      G   /usr/bin/gnome-shell               95MiB |
|    0   N/A  N/A      7327      G   ...AAAAAAAA== --shared-files      150MiB |
|    0   N/A  N/A     33604      G   ...AAAAAAAAA= --shared-files       36MiB |
|    0   N/A  N/A     60502      G   ..._59022.log --shared-files       35MiB |
|    0   N/A  N/A     67654      C   .../git/smot/venv/bin/python    22781MiB |
|    1   N/A  N/A      2618      G   /usr/lib/xorg/Xorg                  4MiB |
|    1   N/A  N/A      2895      G   /usr/bin/gnome-shell                0MiB |
|    1   N/A  N/A      7327      G   ...AAAAAAAA== --shared-files        0MiB |
|    1   N/A  N/A     33604      G   ...AAAAAAAAA= --shared-files        0MiB |
|    1   N/A  N/A     60502      G   ..._59022.log --shared-files        0MiB |
|    1   N/A  N/A     67654      C   .../git/smot/venv/bin/python      405MiB |
+-----------------------------------------------------------------------------+
```

```
(venv)crutcher@heatlamp:~/git/smot$ dpkg -l | grep cudnn
ii  libcudnn8                                  8.0.5.39-1+cuda11.1                                      amd64        cuDNN runtime libraries
ii  libcudnn8-dev                              8.0.5.39-1+cuda11.1                                      amd64        cuDNN development libraries and headers
```

```
(venv)crutcher@heatlamp:~/git/smot$ dpkg -l | grep cuda
ii  cuda                                       11.1.1-1                                                 amd64        CUDA meta-package
ii  cuda-11-1                                  11.1.1-1                                                 amd64        CUDA 11.1 meta-package
ii  cuda-command-line-tools-11-1               11.1.1-1                                                 amd64        CUDA command-line tools
ii  cuda-compiler-11-1                         11.1.1-1                                                 amd64        CUDA compiler
ii  cuda-cudart-11-1                           11.1.74-1                                                amd64        CUDA Runtime native Libraries
ii  cuda-cudart-dev-11-1                       11.1.74-1                                                amd64        CUDA Runtime native dev links, headers
ii  cuda-cuobjdump-11-1                        11.1.74-1                                                amd64        CUDA cuobjdump
ii  cuda-cupti-11-1                            11.1.105-1                                               amd64        CUDA profiling tools runtime libs.
ii  cuda-cupti-dev-11-1                        11.1.105-1                                               amd64        CUDA profiling tools interface.
ii  cuda-demo-suite-11-1                       11.1.74-1                                                amd64        Demo suite for CUDA
ii  cuda-documentation-11-1                    11.1.105-1                                               amd64        CUDA documentation
ii  cuda-driver-dev-11-1                       11.1.74-1                                                amd64        CUDA Driver native dev stub library
ii  cuda-drivers                               455.32.00-1                                              amd64        CUDA Driver meta-package, branch-agnostic
ii  cuda-drivers-455                           455.32.00-1                                              amd64        CUDA Driver meta-package, branch-specific
ii  cuda-gdb-11-1                              11.1.105-1                                               amd64        CUDA-GDB
ii  cuda-libraries-11-1                        11.1.1-1                                                 amd64        CUDA Libraries 11.1 meta-package
ii  cuda-libraries-dev-11-1                    11.1.1-1                                                 amd64        CUDA Libraries 11.1 development meta-package
ii  cuda-memcheck-11-1                         11.1.105-1                                               amd64        CUDA-MEMCHECK
ii  cuda-nsight-11-1                           11.1.105-1                                               amd64        CUDA nsight
ii  cuda-nsight-compute-11-1                   11.1.1-1                                                 amd64        NVIDIA Nsight Compute
ii  cuda-nsight-systems-11-1                   11.1.1-1                                                 amd64        NVIDIA Nsight Systems
ii  cuda-nvcc-11-1                             11.1.105-1                                               amd64        CUDA nvcc
ii  cuda-nvdisasm-11-1                         11.1.74-1                                                amd64        CUDA disassembler
ii  cuda-nvml-dev-11-1                         11.1.74-1                                                amd64        NVML native dev links, headers
ii  cuda-nvprof-11-1                           11.1.105-1                                               amd64        CUDA Profiler tools
ii  cuda-nvprune-11-1                          11.1.74-1                                                amd64        CUDA nvprune
ii  cuda-nvrtc-11-1                            11.1.105-1                                               amd64        NVRTC native runtime libraries
ii  cuda-nvrtc-dev-11-1                        11.1.105-1                                               amd64        NVRTC native dev links, headers
ii  cuda-nvtx-11-1                             11.1.74-1                                                amd64        NVIDIA Tools Extension
ii  cuda-nvvp-11-1                             11.1.105-1                                               amd64        CUDA Profiler tools
ii  cuda-repo-ubuntu2004-11-1-local            11.1.1-455.32.00-1                                       amd64        cuda repository configuration files
ii  cuda-runtime-11-1                          11.1.1-1                                                 amd64        CUDA Runtime 11.1 meta-package
ii  cuda-samples-11-1                          11.1.105-1                                               amd64        CUDA example applications
ii  cuda-sanitizer-11-1                        11.1.105-1                                               amd64        CUDA Sanitizer
ii  cuda-toolkit-11-1                          11.1.1-1                                                 amd64        CUDA Toolkit 11.1 meta-package
ii  cuda-tools-11-1                            11.1.1-1                                                 amd64        CUDA Tools meta-package
ii  cuda-visual-tools-11-1                     11.1.1-1                                                 amd64        CUDA visual tools
ii  libcudart10.1:amd64                        10.1.243-3                                               amd64        NVIDIA CUDA Runtime Library
ii  libcudnn8                                  8.0.5.39-1+cuda11.1                                      amd64        cuDNN runtime libraries
ii  libcudnn8-dev                              8.0.5.39-1+cuda11.1                                      amd64        cuDNN development libraries and headers
ii  nvidia-cuda-dev                            10.1.243-3                                               amd64        NVIDIA CUDA development files
ii  nvidia-cuda-doc                            10.1.243-3                                               all          NVIDIA CUDA and OpenCL documentation
ii  nvidia-cuda-gdb                            10.1.243-3                                               amd64        NVIDIA CUDA Debugger (GDB)
ii  nvidia-cuda-toolkit                        10.1.243-3                                               amd64        NVIDIA CUDA development toolkit

```

```
(venv)crutcher@heatlamp:~/git/smot$ pip show tf-nightly
Name: tf-nightly
Version: 2.5.0.dev20201213
Summary: TensorFlow is an open source machine learning framework for everyone.
Home-page: https://www.tensorflow.org/
Author: Google Inc.
Author-email: packages@tensorflow.org
License: Apache 2.0
Location: /home/crutcher/git/smot/venv/lib/python3.8/site-packages
Requires: tb-nightly, six, tf-estimator-nightly, grpcio, opt-einsum, termcolor, google-pasta, flatbuffers, h5py, astunparse, keras-preprocessing, numpy, wheel, wrapt, protobuf, gast, absl-py, typing-extensions
Required-by: 
```

**Comment by bhack** [comment]
I think that this is a duplicate of https://github.com/tensorflow/tensorflow/issues/44750

**Comment by bayesrule** [comment]
exactly the same issue

**Comment by whtop** [comment]
> **System information**
> 
> * Have I written custom code (as opposed to using a stock example script provided in TensorFlow):yes
> * OS Platform and Distribution (e.g., Linux Ubuntu 16.04):docker
> * TensorFlow installed from (source or binary):pip
> * TensorFlow version (use command below):tf-nightly-gpu         2.5.0.dev20201210
> * Python version:3.6.9
> * CUDA/cuDNN version:11.0/8.0.4
> * GPU model and memory:3090/24GB
> 
> when i inference my tf model,it appears hundreds of warnings like the following image
> about 1 minute later ,the normal inference begin
> i watch the issus and try to used tf2.4re02 but it was useless
> ![image](https://user-images.githubusercontent.com/33931180/101855756-ef43c680-3b9e-11eb-807e-7756853c75ab.png)

Did you solve the issue? If you do, can you share how you do it.

**Comment by sanjoy** [comment]
These warnings are mainly there for diagnostic purposes (so that when a user reports a bug we know that they were using the driver JIT and not ptxas).  For now you can ignore them, and we'll fix the warning to be less verbose / scary.

**Comment by bayesrule** [comment]
@sanjoy but it has been reported performance loss due to this ptxas fatal . e.g. https://dobromyslova.medium.com/making-work-tensorflow-with-nvidia-rtx-3090-on-windows-10-7a38e8e582bf

**Comment by Crispy13** [comment]
I had the same issue. 
Using tensorflow compiled with cuda 11.1 solved the issue.

**Comment by GuillaumeMougeot** [comment]
I had the same issue.
In my case, replacing the ptxas file from cuda 11.0 by the ptxas file from 11.1 (or 11.2) solved the problem. This file can be found in "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.0\bin\" on Windows and in "/usr/local/cuda-11.0/bin/" on Linux. 
(I also have a rtx 3090 with tf 2.4.1)

**Comment by rajasd27** [comment]
> I had the same issue.
> In my case, replacing the ptxas file from cuda 11.0 by the ptxas file from 11.1 (or 11.2) solved the problem. This file can be found in "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.0\bin" on Windows and in "/usr/local/cuda-11.0/bin/" on Linux.
> (I also have a rtx 3090 with tf 2.4.1)

Did you just replace the ptax file or upgrade cuda from 11.0 to 11.1/11.2? @GuillaumeMougeot

**Comment by GuillaumeMougeot** [comment]
> Did you just replace the ptax file or upgrade cuda from 11.0 to 11.1/11.2? @GuillaumeMougeot

I just replaced the ptxas file. After installing cuda 11.0, I had to download cuda 11.2. I then copied the ptxas file from cuda 11.2 into cuda 11.0 and the warnings stopped. @rajasd27

**Comment by ryol8888** [comment]
`I had the same issue.
In my case, replacing the ptxas file from cuda 11.0 by the ptxas file from 11.1 (or 11.2) solved the problem. This file can be found in "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.0\bin" on Windows and in "/usr/local/cuda-11.0/bin/" on Linux.
(I also have a rtx 3090 with tf 2.4.1)
`
It works for me

**Comment by rajasd27** [comment]
> > Did you just replace the ptax file or upgrade cuda from 11.0 to 11.1/11.2? @GuillaumeMougeot
> 
> I just replaced the ptxas file. After installing cuda 11.0, I had to download cuda 11.2. I then copied the ptxas file from cuda 11.2 into cuda 11.0 and the warnings stopped. @rajasd27

Worked for me. I was using tf-gpu docker so that to rebuild a new image by replacing the ptxas file. Thanks a lot!

**Comment by ctc53** [comment]
1. Upgrade to cuda 11.1+ doesn't work for me. There is an error:
    Could not load dynamic library 'libcusolver.so.10'; dlerror: libcusolver.so.10: cannot open shared object file: No such file or directory.
2. Manually copy ptxas in cuda 11.1+ doesn't work either. There would be an exception during evaluation.
3. Still waiting for an update. 
( linux version: linux 5.10.16.arch1-1 / RTX3090/ cuda 11.0.3-1 /cudnn 8.1.0.77-1/ nvidia 460.39-8)

**Comment by bhack** [comment]
TF with new cuda versions is tracked at https://github.com/tensorflow/tensorflow/issues/46093

**Comment by ljburtz** [comment]
I had the same ptax related error when using the Tensorflow provided container (tensorflow/tensorflow:latest-gpu-jupyter -> tensorflow 2.4)

Fixed by using the NVIDIA provided container (nvcr.io/nvidia/tensorflow:21.02-tf2-py3 also tensorflow 2.4)
available at
https://docs.nvidia.com/deeplearning/frameworks/tensorflow-release-notes/running.html#running

**Comment by Chen8023** [comment]
> > I had the same issue.
> > In my case, replacing the ptxas file from cuda 11.0 by the ptxas file from 11.1 (or 11.2) solved the problem. This file can be found in "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.0\bin" on Windows and in "/usr/local/cuda-11.0/bin/" on Linux.
> > (I also have a rtx 3090 with tf 2.4.1)
> 
> It works for me

Thanks！！!It also works for me.

**Comment by rmothukuru** [comment]
@liuxufenfeiya,
Can you please let us know if we can close this issue as it has been resolved? Thanks!

**Comment by disrae** [comment]
> These warnings are mainly there for diagnostic purposes (so that when a user reports a bug we know that they were using the driver JIT and not ptxas). For now you can ignore them, and we'll fix the warning to be less verbose / scary.

I think this issue is causing:
- Model does not begin training until GPU memory is full and then it switches to CPU.
- So this warning prints out until the GPU can't be used and then it stops and the CPU starts.

**Comment by mujiankang** [comment]
> It works for me



> It works for me (just 11.2)

**Comment by Marroh** [comment]
> Try upgrading your cuda version to 11.1

True, just make sure CUDA_HOME and LD_LIBRARY_PATH refer to the path of cuda11.1
